proxy-support/README.md

```diff
@@ -0,0 +1,273 @@
+
+For that, the values of `no_proxy` are separated into two logical parts:
+
+- endpoints which are installation wide like a customer internal container registry or a customer internal network range.
+- endpoints which are cluster specific like the `podCIDR`/`serviceCIDR` or the cluster internal base domain (e.g. `cluster.local`)
+
+> Note: the interpretation of the `no_proxy` variable is very language/tool specific and not every language supports e.g. network ranges.
+> a very details blog post was written by GitLab about [proxy variables].
+
+## Implementation
+
+### Management Cluster
+
+#### during the initial bootstrap
+
+During the initial bootstrap phase of the `MC`, there isn't the entire App Platform running. For that reason it's required to define the proxy configuration multiple times.
+
+#### after all controllers/operators are in place
+
+After the initial bootstrap is done and all GiantSwarm specific controllers and operators are in place to treat the `MC` as a `WC`, [`cluster-apps-operator`] will take over the handling of the installation specific proxy configuration.
+
+### Workload Clusters
+
+The current implemented workflow can be visualized like this:
+
+![](data/workload_cluster_creation.svg)
+
+#### [`cluster-apps-operator`]
+
+As the proxy configuration is something which exists (for now) on a installation base and must be injected on a WC base (to make cluster-lifecycle work), in [PR #290] [`cluster-apps-operator`] got extended to propagate global proxy configuration into workload cluster specific resources.
+
+[`cluster-apps-operator`] can then be configured to accept following configuration as additional input (added as `configmap-values.yaml.patch` and `secret-values.yaml.patch` in the [`config`]-repo):
+
+```yaml
+proxy:
+  noProxy: "172.16.0.0/16,internal.corpdomain.net,example.com"
+  http: "http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+  https: "http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+```
+
+As [`cluster-apps-operator`] already creates a `configmap` for each WC called `<clusterName>-cluster-values`, with the extended proxy implementation, the `cluster` section of the `configmap` now contains a `proxy` sub section:
+
+```yaml
+calico:
+  CIDR: 100.96.0.0/11
+kubernetes:
+  API:
+    clusterIPRange: 10.96.0.0/12
+  DNS:
+    IP: 10.96.0.10
+proxy:
+  noProxy: cluster.local,100.64.0.0/13,100.96.0.0/11,178.170.32.59,172.16.0.0/16,internal.corpdomain.net,example.com,svc,127.0.0.1,localhost
+```
+
+[`cluster-apps-operator`] also creates a `secret` which contains some infrastructure specific values ([for `openstack` and `vcd`](https://github.com/giantswarm/cluster-apps-operator/blob/master/service/controller/resource/clustersecret/desired.go#L42-L65)).
+The `secret` for each WC is called `<clusterName>-cluster-values`.
```

Comment:
Did we split the proxy configuration into 2 places? I think we can put the `noProxy` values into the secret too. Noone will use the noProxy without others and it is not a problem to put a non-sensitive data into a secret. As far as I understand, it will not require any change in other components, only in cluster-apps-operator.

Reply:
yes, currently the configuration is split into `secret/<clustername>-cluster-values` and `configmap/<clustername>-cluster-values`.

Also the configuration for `cluster-apps-operator` is currently done in the `config`-repo in the files:
- `installations/<INSTALLATION_NAME>/apps/cluster-apps-operator/configmap-values.yaml.patch`
- `installations/<INSTALLATION_NAME>/apps/cluster-apps-operator/secret-values.yaml.patch`

Reply:
I've updated the `cluster-apps-operator` PR to safe the `noProxy` configuration in the `secret/<clustername>-cluster-values`. the `configmap/<clustername>-cluster-values` will be unchanged (compared to the current released version of `cluster-apps-operator`)